test/recipes/90-test_external.t

```diff
@@ -11,34 +11,61 @@ use OpenSSL::Test;
 use OpenSSL::Test::Utils;
 use OpenSSL::Test qw/:DEFAULT bldtop_file srctop_file cmdstr/;
 
+my $curtest;
+my $numtests = 0;
+
 setup("test_external");
 
 plan skip_all => "No external tests in this configuration"
     if disabled("external-tests");
 
-if (!$ENV{BORING_RUNNER_DIR}) {
+if ($ENV{BORING_RUNNER_DIR}) {
+    $numtests += 1;
+}
+if ($ENV{KRB5_DIR}) {
+    $numtests += 1;
+}
+
+if ($numtests == 0) {
     plan skip_all => "No external tests have been detected";
 }
 
-plan tests => 1;
+plan tests => $numtests;
 
-indir $ENV{BORING_RUNNER_DIR} => sub {
-    ok(filter_run(cmd(["go", "test", "-shim-path",
-                      bldtop_file("test", "ossl_shim", "ossl_shim"),
-                      "-shim-config",
-                      srctop_file("test", "ossl_shim", "ossl_config.json"),
-                      "-pipe", "-allow-unimplemented"])),
-        "running external tests");
-}, create => 0, cleanup => 0;
+if ($ENV{BORING_RUNNER_DIR}) {
+    $curtest = "boring";
+    indir $ENV{BORING_RUNNER_DIR} => sub {
+        ok(filter_run(cmd(["go", "test", "-shim-path",
+                           bldtop_file("test", "ossl_shim", "ossl_shim"),
+                           "-shim-config",
+                           srctop_file("test", "ossl_shim", "ossl_config.json"),
+                           "-pipe", "-allow-unimplemented"])),
+           "running boringssl tests");
+    }, create => 0, cleanup => 0;
+}
+
+if ($ENV{KRB5_DIR}) {
+    $curtest = "krb5";
+    indir $ENV{KRB5_DIR} => sub {
+        ok(filter_run(cmd(["sh", "-c",
+                           "autoreconf -fiv && " .
+                           "./configure --with-ldap --with-prng-alg=os " .
+                           "--with-crypto-impl=openssl --enable-pkinit " .
+                           "--with-pkinit-crypto-impl=openssl " .
+                           "--with-tls-impl=openssl && " .
```

Comment:
Yup.  And the krb5 configure doesn't seem to have logic specifically to find an openssl at a path, so I think it's the manual CPPFLAGS/LDFLAGS route for us.